### PR TITLE
Fix general warnings and errors in Xcode 9

### LINKED ISSUE #542: Cannot build
While trying to build on XCode 9 beta I received this message:

`'MessageCompletionObject' requires that 'AuthSettings?' inherit from 'Object'`

screenshot 2017-07-25 alle 08 13 51

### PR DESCRIPTION
**Pull request by abintom**
@RocketChat/iOS
Fixes warnings and errors:
1. Fix generics optional error issue in "MessageCompletionObject" closure
2. Optimizes "for if" conditions to "for where"
3. Fixes NotificationCenter observer removal warnings of self on view disappear

Closes #542

### CODE REVIEW

File: Rocket.Chat/Controllers/Chat/ChatViewController.swift [hunk 1]
```diff
@@ -418,10 +418,9 @@ final class ChatViewController: SLKTextViewController {
                     var insert = true
 
                     // swiftlint:disable for_where
```
Comment (rafaelks) [inline Rocket.Chat/Controllers/Chat/ChatViewController.swift]: Can you remove this `swiftlint` comment?

File: Rocket.Chat/Controllers/Chat/ChatDataController.swift [hunk 2]
```diff
@@ -87,12 +87,11 @@ final class ChatDataController {
                     // Check if already contains some separator with this data
                     // swiftlint:disable for_where
```
Comment (rafaelks) [inline Rocket.Chat/Controllers/Chat/ChatDataController.swift]: Can you remove this `swiftlint` comment?

File: Rocket.Chat/Controllers/Chat/ChatViewController.swift (hunk 1 above)
Comment (abintom) [reply Rocket.Chat/Controllers/Chat/ChatViewController.swift]: Done. 👍

File: Rocket.Chat/Controllers/Chat/ChatDataController.swift (hunk 2 above)
Comment (abintom) [reply Rocket.Chat/Controllers/Chat/ChatDataController.swift]: Done. Thank you. Didn't see that 😄

**Comment by rafaelks** [comment]
@abintom Looks like your PR is not rebased to `develop` branch. Can you rebase it before we merge?

**Comment by rafaelks** [comment]
@abintom Thank you!!!

**Pull request merged**